### PR TITLE
kind-conformance-image: use latest kind like the rest of CI

### PR DESCRIPTION
**Pull request by mikedanese**
And find the right kubectl with aquery.

cc @justaugustus @BenTheElder

### CODE REVIEW

File: experiment/kind-conformance-image-e2e.sh [hunk 1]
```diff
@@ -48,7 +39,7 @@ cleanup() {
 install_kind() {
     # install `kind` to tempdir
     TMP_DIR=$(mktemp -d)
-    curl -sLo "${TMP_DIR}/kind" https://github.com/kubernetes-sigs/kind/releases/download/${STABLE_KIND_VERSION}/kind-linux-amd64
+    curl -sLo "${TMP_DIR}/kind" https://kind.sigs.k8s.io/dl/latest/kind-linux-amd64
```
Comment (mikedanese) [inline experiment/kind-conformance-image-e2e.sh:42]: @BenTheElder wdyt with aligning with other CI jobs here? Alternatively, we can tag a new kind release.

**Comment by BenTheElder** [comment]
FYI this job is not blocking for a reason, it's something the sonobuoy folks more or less set up to get coverage for the e2e.test wrapper image but it's not something anyone has been looking after properly. Upstream CI does not use the e2e wrapper image otherwise for various reasons.

I would be in favor of demoting this to CI given the limited need and the lack of upkeep.

File: experiment/kind-conformance-image-e2e.sh (hunk 1 above)
Comment (BenTheElder) [reply experiment/kind-conformance-image-e2e.sh:42]: There was a new release tagged last night, but switching to latest here is reasonable.

I think this job should be demoted out of release blocking and presubmit given the lack of maintainership, it's frequently failing.
https://testgrid.k8s.io/conformance-all#conformance-image,%20master%20(dev)

For some additional context:

All of the other jobs using this `latest` binary the KIND maintainers actively support and we block kind PRs on running mirrors of the jobs. 

The `latest` (nightly / continuous) binary is only supported for upstream k/k testing where we need to move quicker to work around upstream breaking changes like this.

This image is really only consumed by sonobuoy, which is not upstream.

**Comment by mikedanese** [comment]
@bentheelder thanks. Looks like it only runs if conformance changed:

https://github.com/kubernetes/test-infra/blob/4612a9ca313b199daa4fc782863ccb8d8723a482/config/jobs/kubernetes/sig-testing/conformance-e2e.yaml#L7-L11

cc @dims

File: experiment/kind-conformance-image-e2e.sh (hunk 1 above)
Comment (BenTheElder) [reply experiment/kind-conformance-image-e2e.sh:42]: Actually, it appears the CI version of this job is not even in release informing currently, let alone blocking.

**Comment by BenTheElder** [comment]
thanks @mikedanese :+1:

**Pull request merged**